lib/dat-worker-pool/runner.rb

```diff
@@ -78,27 +97,44 @@ def build_worker
     # ensures if the hard shutdown is raised and not rescued (for example, in
     # the workers ensure), then it won't cause the forced shutdown to end
     # prematurely
-    def force_shutdown(timeout, backtrace)
-      error = ShutdownError.new("Timed out shutting down (#{timeout} seconds).")
-      error.set_backtrace(backtrace)
+    def forced_shutdown(orig_exception, timeout, &backtrace_proc)
+      error = build_forced_shutdown_error(orig_exception, timeout, backtrace_proc)
       until @workers.empty?
         worker = @workers.first
         worker.dwp_raise(error)
         worker.dwp_join rescue false
```

Comment:
@jcredding we discussed this is a `rescue StandardError` thing right?  Do you want to be explicit here and only rescue standard errors on join?